scipy/stats/tests/test_stats.py

```diff
@@ -7249,6 +7294,88 @@ def test_inf_values(self):
             assert_equal(
                 stats.wasserstein_distance([1, 2, np.inf], [np.inf, 1]),
                 np.nan)
+        uv, vv, uw = [[1, 1], [2, 1]], [[np.inf, -np.inf]], [1, 1]
+        distance = stats.wasserstein_distance(uv, vv, uw)
+        assert_equal(distance, np.inf)
+        with suppress_warnings() as sup:
+            sup.record(RuntimeWarning, "invalid value*")
+            uv, vv = [[np.inf, np.inf]], [[np.inf, -np.inf]]
+            distance = stats.wasserstein_distance(uv, vv)
+            assert_equal(distance, np.nan)
+
+    def test_multi_dim(self):
+        # Adding dimension on distributions do not affect the result
+        uv_1, vv_1 = [[1, 0], [2, 0], [3, 0]], [[-1, 0], [5, 0], [1, 0]]
+        distance_1 = stats.wasserstein_distance(uv_1, vv_1)
+        uv_2, vv_2 = [1, 2, 3], [-1, 5, 1]
+        distance_2 = stats.wasserstein_distance(uv_2, vv_2)
+        assert_almost_equal(distance_1, distance_2)
+        uv_1, vv_1 = [[0, 1, 2]], [[0, 1, 1]]
+        distance_1 = stats.wasserstein_distance(uv_1, vv_1)
+        uv_2, vv_2 = [[1, 2]], [[1, 1]]
+        distance_2 = stats.wasserstein_distance(uv_2, vv_2)
+        assert_almost_equal(distance_1, distance_2)
+        uv_1 = [[-1, 0], [0, 0], [4, 0], [8, 0]]
+        vv_1 = [[-1, 0], [3, 0], [2, 0], [0, 0]]
+        uw_1, vw_1 = [1, 3, 5, 7], [2, 4, 6, 8]
+        distance_1 = stats.wasserstein_distance(uv_1, vv_1, uw_1, vw_1)
+        uv_2, vv_2 = [[-1], [0], [4], [8]], [[-1], [3], [2], [0]]
+        uw_2, vw_2 = [1, 3, 5, 7], [2, 4, 6, 8]
+        distance_2 = stats.wasserstein_distance(uv_2, vv_2, uw_2, vw_2)
+        assert_almost_equal(distance_1, distance_2)
+
+    @pytest.mark.parametrize('input_dim', (2, 3, 7, 15, 60))
+    @pytest.mark.parametrize('data_volumn', ((3, 5), (12, 16), (20, 20),
+                                             (1, 200), (15, 15)))
+    def test_shift_nD(self, input_dim, data_volumn):
+        # Moving across n-dimensional space does not change the outcome.
+        rng = np.random.default_rng(734825773)
+        uu = rng.random((data_volumn[0], input_dim))
+        vv = rng.random((data_volumn[1], input_dim))
+        expected = stats.wasserstein_distance(uu, vv)
+        shift = rng.random(input_dim)
+        uu_shift, vv_shift = uu + shift, vv + shift
+        shift_result = stats.wasserstein_distance(uu_shift, vv_shift)
+        assert_almost_equal(expected, shift_result)
+
+    @pytest.mark.parametrize('input_dim', (3, 4, 5, 22, 40))
```

Comment:
```suggestion
    @pytest.mark.parametrize('input_dim', (3, 4, 5, 22))
```
I think this is causing most of the CI failures on Azure Pipelines.
```
FAILED ..\..\stats\tests\test_stats.py::TestWassersteinDistance::test_rotation_reflection_nD[data_volumn0-40]
FAILED ..\..\stats\tests\test_stats.py::TestWassersteinDistance::test_rotation_reflection_nD[data_volumn1-40]
FAILED ..\..\stats\tests\test_stats.py::TestWassersteinDistance::test_rotation_reflection_nD[data_volumn3-40]
FAILED ..\..\stats\tests\test_stats.py::TestWassersteinDistance::test_rotation_reflection_nD[data_volumn2-40]
FAILED ..\..\stats\tests\test_stats.py::TestWassersteinDistance::test_rotation_reflection_nD[data_volumn4-40]
```
Specifically,
```
File "C:\hostedtoolcache\windows\Python\3.9.13\x86\lib\site-packages\scipy\stats\tests\test_stats.py", line 7373, in test_rotation_reflection_nD
    rotation_matrix = np.linalg.multi_dot(rotations)
  File "<__array_function__ internals>", line 180, in multi_dot
  File "C:\hostedtoolcache\windows\Python\3.9.13\x86\lib\site-packages\numpy\linalg\linalg.py", line 2726, in multi_dot
    order = _multi_dot_matrix_chain_order(arrays)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x86\lib\site-packages\numpy\linalg\linalg.py", line 2792, in _multi_dot_matrix_chain_order
    q = m[i, k] + m[k+1, j] + p[i]*p[k+1]*p[j+1]

+++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++
.....[gw5] node down: Not properly terminated

```
It's probably just that the AZP resources are insufficient. (Do you expect that to be a demanding problem for `np.linalg.multi_dot`? If not, it might be worth filing a separate bug report.) In any case, it does not indicate a problem with `wasserstein_distance`, so let's just avoid it.

Reply:
> Do you expect that to be a demanding problem for `np.linalg.multi_dot`
No, I don't foresee that. I will remove that in the next update.